src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs

```diff
@@ -493,6 +497,10 @@ private void UpdatePackageList(IEnumerable<PackageItemViewModel> packages, bool
                         _selectedCount = package.IsSelected ? _selectedCount + 1 : _selectedCount;
                     }
 
+                    // update the top-level and transitive package counts
+                    TopLevelPackageCount = PackageItems.Count(p => p.PackageLevel == PackageLevel.TopLevel);
+                    TransitivePackageCount = PackageItems.Count(p => p.PackageLevel == PackageLevel.Transitive);
```

Comment:
``` suggestion
                    TransitivePackageCount = FilteredItemsCount - TopLevelPackageCount ;
```

Given that [`Enumerable.Count`](https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.count) is optimized if source is of type `ICollection`, Can we compute the transitive packages count using above code?

Looks like `PackageItems` is an Array internally. Would it be possible to debug [code](https://docs.microsoft.com/en-us/visualstudio/debugger/how-to-debug-dotnet-framework-source?view=vs-2022) to check if the code is going through optimized code path or not?
https://github.com/NuGet/NuGet.Client/pull/4490/files#diff-40ccb6a5da9619e302036d6319a72cdd3b485457bf337f1d8e9c882ace1b47e3R136

Reply:
`PackageItems` is an array. See https://github.com/NuGet/NuGet.Client/blob/15d4dfc212caad3648099a20dc20eaab4b8272c9/src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs#L136

This change or actual PR code change, LGTM.

Reply:
I agree that `PackageItems` is an array. My point is that we can skip iterating through the collection to identify transitive packages because we already know `total package items` and `top level packages count`. Hence, I suggested the code below to avoid iterating through the collection.

```cs
TransitivePackageCount = FilteredItemsCount - TopLevelPackageCount ;
```

 This suggestion will not work if `PackageItems` collection contains more than 2 categories of packages such as top level and transitive packages.

Reply:
Thanks for the suggestion @kartheekp-ms! I'm in favor of the explicit counting of the items by their level type to make the code less brittle to future changes. I see how doing math for the second count saves the trouble of walking the list to count the items, but the gain is expected to be minimal.